### PR TITLE
fix(pr): monorepo changelog with sourceDirectory

### PR DESCRIPTION
**Pull request by zanitete**
take sourceDirectory into account when removing duplicates

## Documentation

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — CHANGES_REQUESTED** [review]
Please add a test to ensure we don't regress in future

File: lib/workers/pr/index.ts [hunk 1]
```diff
@@ -318,10 +318,17 @@ export async function ensurePr(
 
   config.hasReleaseNotes = config.upgrades.some((upg) => upg.hasReleaseNotes);
 
+  function getReleaseNotesSourceWithSourceDirectory(
+    upgrade: BranchUpgradeConfig
+  ): string {
+    return `${
+      upgrade.releases?.[0]?.releaseNotes?.notesSourceUrl || upgrade.sourceUrl
+    }${upgrade.sourceDirectory ? `:${upgrade.sourceDirectory}` : ''}`;
```
Comment (viceice) [inline lib/workers/pr/index.ts:326]: `sourceDirectory` should only be used with `sourceUrl`

**Review by viceice — COMMENTED** [review]
I think this needs a test case

**Review by viceice — DISMISSED** [review]

**Comment by rarkins** [comment]
@zanitete please update the branch so that status checks can pass

**Review by rarkins — DISMISSED** [review]

**Review by viceice — DISMISSED** [review]

**Comment by rarkins** [comment]
@zanitete needs updating from main branch

**Comment by zanitete** [comment]
👍 done!

**Review by viceice — APPROVED** [review]

**Comment by viceice** [comment]
> 👍 done!

https://github.com/renovatebot/renovate/blob/main/.github/contributing.md#re-requesting-a-review

**Pull request merged**